internal/pkg/service/buffer/sink/tablesink/storage/level/local/writer/allocate/config.go

```diff
@@ -0,0 +1,37 @@
+package allocate
+
+import "github.com/c2h5oh/datasize"
+
+// Config configures allocation of the disk space for file slices.
+type Config struct {
+	// Enabled enables disk space allocation.
+	Enabled bool `json:"enabled" configKey:"enabled" configUsage:"Allocate disk space for each slice."`
+	// Size of the first slice in a sink, or the size of each slice if DynamicSize is disabled.
+	Size datasize.ByteSize `json:"size" configKey:"size" validate:"required" configUsage:"Size of allocated disk space for a slice."`
+	// SizePercent enables dynamic size of allocated disk space.
+	// If enabled, the size of the slice will be % from the average slice size.
+	// The size of the first slice in a sink will be Size.
+	SizePercent int `json:"sizePercent" configKey:"sizePercent" validate:"min=100,max=500" configUsage:"Allocate disk space as % from the previous slice size."`
+}
```

Comment:
Disk allocation config is now part of the `allocate` package. I created the new struct.